msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```
@@ -231,15 +318,168 @@ public static IMultipleAccountPublicClientApplication createMultipleAccountPubli
+     *                             {@link PublicClientApplication}. Cannot be null.
+     *                             <p>
+     *                             Note: The {@link Context} should be the application context instead of the running activity's context, which could potentially make the sdk hold a
+     *                             strong reference to the activity, thus preventing correct garbage collection and causing bugs.
+     *                             </p>
+     * @param configFileResourceId The resource ID of the raw file containing the JSON configuration for the PublicClientApplication
+     * @return An instance of ISingleAccountPublicClientApplication
+     * @see <a href="https://developer.android.com/guide/topics/resources/providing-resources">Android app resource overview</a>
+     * @see <a href="https://github.com/AzureAD/microsoft-authentication-library-for-android/wiki">MSAL Github Wiki</a>
+     */
+    @WorkerThread
+    public static ISingleAccountPublicClientApplication createSingleAccountPublicClientApplication(@NonNull final Context context,
+                                                                                                       final int configFileResourceId) throws InterruptedException, MsalException {
+        if (context == null) {
+            throw new IllegalArgumentException("context is null.");
+        }
+
+        final PublicClientApplicationConfiguration configuration = loadConfiguration(context, configFileResourceId);
+
+        IPublicClientApplication application = createPublicClientApplication(context, configuration);
+        if(application instanceof ISingleAccountPublicClientApplication){
+            return (ISingleAccountPublicClientApplication) application;
+        }else{
+            if(configuration.mAccountMode != AccountMode.SINGLE){
+                throw new MsalClientException("AccountMode in configuration is not set to single");
+            }
+            return null;
```

Here as well, better to throw an exception